### PR TITLE
Fix hovering over Sankey node only fully highlights first trace

### LINKED ISSUE #6328: Node hover effect not working correctly in Sankey plots
There seems to be an issue with Sankey plots' nodes' hover effect. When hovering over a node, not all traces entering and exiting that node are fully highlighted.

reproducible example: https://codepen.io/esiivola/pen/ZEovKvN
Instructions: Place your mouse over the node with the title "D". "trace 1" is now highlighted and "trace 2" is highlighted from "H" to "D" and from "D" to "F", but not from "F" to "B". 

Correct behavior should be that either traces 1 & 2 are fully highlighted OR only the segments from "H" to "D" and from "D" to "F" 

This issue has been discussed here (https://community.plotly.com/t/sankey-node-hover-issue/28324) as well but I didn't find a solution or bug ticket.

### PR DESCRIPTION
**Pull request by DominicWuest**
Fixes #6328.

The issue was that only the label of the first link was considered when recursively highlighting traces.

### CODE REVIEW

File: src/traces/sankey/plot.js [hunk 1]
```diff
@@ -62,24 +62,25 @@ function nodeNonHoveredStyle(sankeyNode, d, sankey) {
...
-            .filter(function(l) {return l.link.label === label;})
-            .style('fill-opacity', function(l) {
-                if(!l.link.concentrationscale) {
-                    return 0.4;
-                }
-            });
-    }
+    sankeyLink.each(function(curLink) {
+        var label = curLink.link.label;
+        if(label) {
```
Comment (archmoj) [inline src/traces/sankey/plot.js:73]:
```suggestion
        if(label !== undefined) {
```

File: src/traces/sankey/plot.js [hunk 2]
```diff
@@ -90,15 +91,17 @@ function linkHoveredStyle(d, sankey, visitNodes, sankeyLink) {
...
-    if(label) {
-        ownTrace(sankey, d)
-            .selectAll('.' + cn.sankeyLink)
-            .filter(function(l) {return l.link.label === label;})
-            .style('fill-opacity', function(d) {return d.tinyColorAlpha;});
-    }
+
+    sankeyLink.each(function(curLink) {
+        var label = curLink.link.label;
+        if(label) {
```
Comment (archmoj) [inline src/traces/sankey/plot.js:98]:
```suggestion
        if(label !== undefined) {
```

**Comment by DominicWuest** [comment]
Incorporated your suggestions @archmoj! :)

**Comment by archmoj** [comment]
Please add a test case in `test/jasmine/tests/sankey_test.js` in the hover interaction block and this PR should be ready to go.
You can run 
```sh
 npm run test-jasmine sankey
```
To run just the `sankey` test.
Thank you!

File: draftlogs/6799_fix.md [hunk 3]
```diff
@@ -0,0 +1 @@
+ -  Fix hovering over Sankey node only fully highlights first trace [[#6799](https://github.com/plotly/plotly.js/pull/6799)]
```
Comment (archmoj) [inline draftlogs/6799_fix.md:1]:
```suggestion
 -  Fix hovering over Sankey node only fully highlights first trace [[#6799](https://github.com/plotly/plotly.js/pull/6799)], with thanks to @DominicWuest for the contribution!
 ```

**Comment by DominicWuest** [comment]
Hmm interesting, the tests pass on my end... But only when the browser is in full-screen. I hover over the right-most node, maybe that is not visible or something?
I'll hover over the left-most node instead then...

File: draftlogs/6799_fix.md (hunk 3 above)
Comment (DominicWuest) [reply draftlogs/6799_fix.md:1]: :)

**Comment by DominicWuest** [comment]
Should work now! :)

**Comment by archmoj** [comment]
Great PR. :trophy: 
:dancer:

**Pull request merged**